tests/test_runner.py

```diff
@@ -589,14 +591,19 @@ def test_invalid_sleep():
                 raise
 
 def test_snoop():
+    filename = default_filename()
     code = dedent(
         """
-    x = 5
-    x *= 2
+    def double(x):
+        return 2*x
+
+    double(5)
         """)
     check_simple(code, [
-        ('output', {'parts': [{'text': '','type': 'snoop'}]}),
-        ('output', {'parts': [{'text': '    2 | x = 5\n', 'type': 'snoop'}]}),
-        ('output', {'parts': [{'text': '    3 | x *= 2\n', 'type': 'snoop'}]}),
-        ('output', {'parts': [{'text': ' ...... x = 10\n', 'type': 'snoop'}]})
-    ], mode="snoop")
+        ('output', {'parts': [{'type': 'snoop', 'text': ''}]}),
```

Comment:
Any idea why this is here? Particularly why it's not combined with the others?

Reply:
Maybe due to an external flush call on the output buffer that is not captured by the stream itself? Such as the flush call in runner.callback?

Reply:
Oh right, it's a separate event, not an output part.

Try adding `runner.reset()` (maybe even inside `check_simple`) before running the code.

Reply:
Yes, at some point an empty string is written to the buffer. Afterwards, the output buffer is directly flushed (so not via the SysStream instance), causing the empty string to be written. Then, all snoop writes occur as normal. Question is now why this flush is called.

Reply:
Probably because the output type is now "snoop", so the callback method in python_runner is triggered causing a direct flush?